osu.Server.Queues.ScoreStatisticsProcessor/Commands/Score/ChangeModMultiplierCommand.cs

```diff
@@ -0,0 +1,163 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Dapper;
+using JetBrains.Annotations;
+using McMaster.Extensions.CommandLineUtils;
+using MySqlConnector;
+using osu.Server.QueueProcessor;
+
+namespace osu.Server.Queues.ScoreStatisticsProcessor.Commands.Score
+{
+    [Command("change-mod-multiplier", Description = "Changes a mod's multiplier globally by adjusting all relevant scores' totals.")]
+    public class ChangeModMultiplierCommand
+    {
+        [Option(CommandOptionType.SingleValue, Template = "-r|--ruleset-id", Description = "Required. The ID of the ruleset for the mod whose multiplier is being adjusted.")]
+        public int? RulesetId { get; set; }
+
+        [Option(CommandOptionType.SingleValue, Template = "-m|--mod", Description = "Required. The acronym of the mod whose multiplier is being adjusted.")]
+        public string? ModAcronym { get; set; }
+
+        [Option(CommandOptionType.SingleValue, Template = "--old", Description = "Required. The old multiplier of the mod being adjusted.")]
+        public double? OldMultiplier { get; set; }
+
+        [Option(CommandOptionType.SingleValue, Template = "--new", Description = "Required. The new multiplier of the mod being adjusted.")]
+        public double? NewMultiplier { get; set; }
+
+        [Option(CommandOptionType.SingleValue, Template = "--start-id", Description = "The ID of the `scores` table row to start processing from.")]
+        public ulong? StartId { get; set; }
+
+        [Option(CommandOptionType.SingleOrNoValue, Template = "--batch-size", Description = "The maximum number of scores to fetch in each batch.")]
+        public int BatchSize { get; set; } = 5000;
+
+        [Option(CommandOptionType.SingleOrNoValue, Template = "--dry-run", Description = "Do not actually change any score totals, just display what would be done.")]
+        public bool DryRun { get; set; }
+
+        private readonly StringBuilder sqlBuffer = new StringBuilder();
+        private readonly ElasticQueuePusher elasticQueueProcessor = new ElasticQueuePusher();
+        private readonly HashSet<ElasticQueuePusher.ElasticScoreItem> elasticItems = new HashSet<ElasticQueuePusher.ElasticScoreItem>();
+
+        [UsedImplicitly]
+        public async Task<int> OnExecuteAsync(CommandLineApplication app, CancellationToken cancellationToken)
+        {
+            if (RulesetId == null
+                || string.IsNullOrEmpty(ModAcronym)
+                || OldMultiplier == null
+                || NewMultiplier == null)
+            {
+                await Console.Error.WriteLineAsync("One or more required parameters is missing.");

```

Comment:
Kinda weird to have optional arguments which are required (usually I'd use standard arguments in place of this, but I guess your goal is to have things named?)

Reply:
Yeah I didn't want to have them purely positional to prevent screw-ups from swapping them or something.